src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -269,6 +270,27 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_SetThreadProvis
     return MapOpenThreadError(otErr);
 }
 
+template <class ImplClass>
+CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_SetThreadProvision(const uint8_t * operationalDataset,
+                                                                                    uint32_t operationalDatasetLen)
+{
+    otError otErr = OT_ERROR_FAILED;
+    (void) operationalDatasetLen;
+
+    // Set the dataset as the active dataset for the node.
+    Impl()->LockThreadStack();
+    otErr = otDatasetSetActiveTlvs(mOTInst, operationalDataset);
+    Impl()->UnlockThreadStack();
+
+    // post an event alerting other subsystems about change in provisioning state
+    ChipDeviceEvent event;
+    event.Type                                           = DeviceEventType::kServiceProvisioningChange;
+    event.ServiceProvisioningChange.IsServiceProvisioned = true;
+    PlatformMgr().PostEvent(&event);
```

Comment:
Is this necessary? Are there Service Provisioning in CHIP?

Reply:
+1 `NetworkProvisioning::ConnectivityHandler` has similar logic. It waits for the SLAAC address to be allocated so that the device can be reached within and outside the Thread network. That shall be a better event to listen for.